solr/core/src/test/org/apache/solr/cli/AuthToolTest.java

```diff
@@ -64,15 +64,15 @@ public void testEnableAuth() throws Exception {
     String[] args = {
       "auth",
       "enable",
-      "-zkHost",
+      "-z",
```

Comment:
[0] You've put in a lot of effort in this PR to make sure that we (at least temporarily) maintain backcompat with older options.  Given all that work, it seems a shame to not validate it.

One way we could do that is by randomizing which form of different options gets used in these tests.  Some of the more common CLI parameters could be replaced by calls to a method like:

```
public void getZooKeeperParameter() {
    if (rarely()) return "-z";
    if (rarely()) return "-zkhost";
    return "--zk-host";
}
```

Not important enough to require on this PR, but worth considering if you like the idea.

Reply:
See SOLR-17344 - we can discuss this issue further there.